### DISCUSSION TITLE
Colorizer Tool

### DISCUSSION THREAD

**Discussion by ElectronicsArchiver** [franksouza183/Evolvere-Icons#121] [Show and tell]
<br>

[Dev Branch] | [Draft PR]

<br>

## Things Done

- Documentation

- Prototype

- Example

<br>

## Todo

- Refactor Protoype Into Clean Implementation

  *Could also be done in subsequent PRs~*

<br>

[Dev Branch]: https://github.com/ElectronicsArchiver/Evolvere-Icons/tree/Tools
[Draft PR]: https://github.com/franksouza183/Evolvere-Icons/pull/122

**Comment by franksouza183** [comment]
It's not working. It seems to be reading the template and converting but for some reason the build directory contains the icons with the same colors as the default. Maybe something at the time of replacing the files or it's just using default values? Also, an error occurred when trying to use 'none' as background color: `「 Background » Color 」 'none' is not a hex color string`

**Reply by ElectronicsArchiver** [reply]
Will check out the conversion problem.

For the `none` thing:
Oh yeah, forgot to add it as an allowed 
option for parsing, will adjust that.

**Reply by ElectronicsArchiver** [reply]
Now any case can be used~

**Comment by ElectronicsArchiver** [comment]
The colorization should now work:

( Added `Resources/Examples/Exotic.yaml` with uncommon colors as a test template )

![ItWorks](https://user-images.githubusercontent.com/85485984/167098853-f4cf7a78-d8a7-438e-96ea-a3b252cdadc6.png)

`None` can now be specified, though
lowercase is still not allowed, will
change that with future refactorings.

**Comment by ElectronicsArchiver** [comment]
Gonna open a `Draft PR` so you can more easily see changes~

-> #122

**Comment by franksouza183** [comment]
Amazing work!! Just one thing before merge: the value 'none' should only add the 'opacity' without change the .ColorScheme-Background color value. Or you can just use any valid color instead '#Null' since this breaks svg syntax. Another thing is that 'none' as value should only be possible on background, the other should use a valid color, and if invalid, just use the default color.

**Reply by ElectronicsArchiver** [reply]
Right, sorry forgot about that~

**Comment by franksouza183** [comment]
Ok, I've thinking about the exclusivity on background, if possible, extend it to foreground too, in that way we can make nice monochrome icons only with contourns if foreground is transparent.

**Reply by ElectronicsArchiver** [reply]
Are you talking about having the `none` option for 
`foreground` as well? If so, then that already works.

**Reply by franksouza183** [reply]
> Are you talking about having the `none` option for `foreground` as well? If so, then that already works.

yes, restricted to background and foreground.

**Comment by franksouza183** [comment]
And here some to do, maybe I can get into deno and do it by my self sometime if you can't:

- Use template name as theme name;
- The conversion process must take place in the user's context as a security measure.
- After conversion or before, ask the user whether to install locally (~/.local/share/icons) or system-wide (/usr/share/icons).

**Reply by ElectronicsArchiver** [reply]
> Use the name of the template as the name of the new theme;

I'd recommend to use a property in the theme instead
so you don't restrict one with the other.

> The conversion process must take place in the user's context as a security measure.

Not quite sure what you mean~

> After conversion or before, ask the user whether to install locally (~/.local/share/icons) or system-wide (/usr/share/icons).

I'd recomment using a separate script that can utilize the first or use the original icons

**Comment by franksouza183** [comment]
> I'd recommend to use a property in the theme instead so you don't restrict one with the other.

wonderful

> Not quite sure what you mean~

Use a directory somewhere in user's home as build dir, maybe ~/.temp or ~/.evolvere-colorizer-build?

> I'd recomment using a separate script that can utilize the first or use the original icons

You should know what the best way xD. Would be nice an info at the end recommending choose the new theme via theming tool

**Reply by ElectronicsArchiver** [reply]
> Use a directory somewhere in user's home as build dir, maybe ~/.temp or ~/.evolvere-colorizer-build?

You could have a command line option I guess~
Something like `--build-folder`

**Reply by ElectronicsArchiver** [reply]
Or you could treat the template as a build
config and have the path be set in there.

```yaml

Name : Red Evolvere
Build Folder: ~/.MyThemes/Red

Foreground:
    Color : '333333'
    Alpha : 0.3

```

**Reply by franksouza183** [reply]
First option, `--build-folder` since the build directory isn't directly related to the theme itself. And just use some default dir on home dir in case that is not specified

**Reply by ElectronicsArchiver** [reply]
> First option, `--build-folder` since the build directory isn't directly related to the theme itself. And just use some default dir on home dir in case that is not specified

Can we save the discussed additions for the next PR though, I'd would like to have the basic version committed first~

**Comment by franksouza183** [comment]
There's something wrong in the svg code, I'm checking...

[edit] Found it:

`<?xml version="1" encoding="UTF-8"?>`, where `version` should be `1.0`

[edit] Issue opened #123

**Comment by franksouza183** [comment]
I've invite you as collaborator, I think is better we deal with enhancement/bugfixes trough issues
